### ISSUE TITLE
yts: download links are invalid

### ISSUE THREAD

**Issue by ngosang** [Jackett/Jackett#12494]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
latest
```

### Description

Any search

### Logged Error Messages

.

**Comment by garfield69** [comment]
Both .torrent and magnets are working for me with the YTS indexer.
Problem at your end?

**Comment by ngosang** [comment]
The issue is with the alternative link https://yts.unblockninja.com/ others are fine.

**Comment by garfield69** [comment]
Comparing the returned json torrent set from each of the site domains reveals the following:
https://yts.mx/api/v2/list_movies.json
```json
{"url":"https:\/\/yts.mx\/torrent\/download\/7046237120DFA617E8AAA15695E6DD510C89E49C","hash":"7046237120DFA617E8AAA15695E6DD510C89E49C","quality":"720p","type":"bluray","seeds":0,"peers":0,"size":"1.05 GB","size_bytes":1127428915,"date_uploaded":"2021-11-01 14:28:23","date_uploaded_unix":1635773303}
```
https://yts.nocensor.biz/api/v2/list_movies.json
```json
{"url":"https:\/\/yts.mx\/torrent\/download\/7046237120DFA617E8AAA15695E6DD510C89E49C","hash":"7046237120DFA617E8AAA15695E6DD510C89E49C","quality":"720p","type":"bluray","seeds":0,"peers":0,"size":"1.05 GB","size_bytes":1127428915,"date_uploaded":"2021-11-01 14:28:23","date_uploaded_unix":1635773303}
```
https://yts.unblockit.kim/api/v2/list_movies.json
```json
{"url":"https:\/\/yts.unblockit.kim\/torrent\/download\/7046237120DFA617E8AAA15695E6DD510C89E49C","hash":"7046237120DFA617E8AAA15695E6DD510C89E49C","quality":"720p","type":"bluray","seeds":0,"peers":0,"size":"1.05 GB","size_bytes":1127428915,"date_uploaded":"2021-11-01 14:28:23","date_uploaded_unix":1635773303}
```
https://yts.unblockninja.com/api/v2/list_movies.json
```json
{"url":"https:\/\/yts.mx\/torrent\/download\/7046237120DFA617E8AAA15695E6DD510C89E49C","hash":"7046237120DFA617E8AAA15695E6DD510C89E49C","quality":"720p","type":"bluray","seeds":0,"peers":0,"size":"1.05 GB","size_bytes":1127428915,"date_uploaded":"2021-11-01 14:28:23","date_uploaded_unix":1635773303}
```

In summary, the torrent details are identical between YTS.mx and yts.nocensor.biz and yts.unblockninja.com
and the only difference is that yts.unblockit.kim uses its own domain in the url.

I should note that all downloads work for my indexer using any of the domains as primary sitelinks in my tests.

Whatever you issue is I doubt it has anything to do with the indexer code.

**Comment by ilike2burnthing** [comment]
If someone is using a proxy though, then chances are they can't access the official site, so trying to download a torrent file from the domain is going to fail.

We could try replacing `yts.mx` with the sitelink. Something like:
```
    sitelink_trimmed:
      text: {{ .Config.sitelink }}
      filters:
        - name: replace
          args: ["https://", ""]
        - name: replace
          args: ["/", ""]
    download:
      selector: url
      filters:
        - name: replace
          args: ["yts.mx", "{{ .Result.sitelink_trimmed }}"] # fix for 12494
```
(untested)

**Comment by garfield69** [comment]
I suppose so. The original yts.cs did have this routine
```C#
        private Uri UrlWithSiteLink(string url)
        {
            var siteLinkUri = new Uri(SiteLink);
            var builder = new UriBuilder(url)
            {
                Scheme = siteLinkUri.Scheme,
                Host = siteLinkUri.Host
            };
            return builder.Uri;
        }
```
which it called when setting release.  Link, Details and Poster
I tested your suggestion
I had to add quotes around the string for sitelink_trimmed, and tested each of the alternate sitelinks.
Worked for all but yts.nocensor.biz, which returns a BencodeParser failure.

But then again, this is exactly what happens when I use yts.nocensor.biz with the old C# indexer so, parity.

I'll add your enhancement since it makes sense to retain parity between old C# and new yaml
